index.js

```diff
@@ -290,21 +289,29 @@ var MongodbDriver = Base.extend({
       };
 
       // Get a connection to mongo
-      this.connection.connect(this.connectionString, function(err, db) {
+      this.connection.connect(function(err, mongoClient) {
 
         if(err) {
           prCB(err);
         }
 
+        // New in Mongo 3
+        // Uses database name from connection string
+        const db = mongoClient.db();
+
         // Callback function to return mongo records
         var callbackFunction = function(err, data) {
 
           if(err) {
             prCB(err);
           }
 
+          // with close(force: true), get db destroyed error
+          // with close(force: false), times out for .collecions
+          // without close, hangs
+          // mongoClient.close(true, function() {
           prCB(null, data);
-          db.close();
```

Comment:
are you sure that it is safe to remove the close call without replacing it? This will cause migrations to never end since the connection is not being released.

Reply:
I would like to close the connection, but I have run into problems when closing it at the end of this function (see comments above that line).

Reply:
mhm that is weird. it shouldn't time out, any idea yet why this happens? However I did not introduce the direct close in this driver, but I know it was introduced b/c there were problems as well if you do not close it after an action. All other drivers close on the dedicated close function of the driver after db-migrate is done.

Reply:
I feel like there's a race condition. When I do close it, other problems pop up later on in the execution

Reply:
Were you able to close the connections? I have been trying to do the same but no luck.

Reply:
I did some progress here. When I re-write the 
```js
      if (!this.connection.isConnected()) {
          try {
            await this.connection.connect();
          } catch (err) {
            return prCB(err);
          }
      }
      const db = this.connection.db();
```

Then I can call `db.close();` without tests hanging. Only thing I'm looking into is why 2 tests are failing when I'm doing this.

Because according to [this](https://jira.mongodb.org/browse/NODE-1868?focusedCommentId=2149851&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2149851) we shouldn't call `connect` multiple times.

This change also removed all the `the options [dbName] is not supported` warnings I received when running the tests.

I can dig more into it but there are some things I'd like to understand before I'm spending more time on this PR:
1. Are you OK @wzrdtales that we moved away from `vows` to `@hapi/lab`? I personally think it's a nice improvement but you have the last say as you're the nr. 1 maintainer 🙂 
2. We have a property "connection": https://github.com/db-migrate/mongodb/blob/60f0c04bb2db085c361f609487f136346115dbcc/index.js#L13
But actually we initialize the property with `MongoClient` https://github.com/db-migrate/mongodb/blob/60f0c04bb2db085c361f609487f136346115dbcc/index.js#L554-L555
As a newbie to this lib I found this very confusing. Are you OK if I rename the class attribute?
3. It seems like this PR can't be merged because commitlint is complaining for some commits. Could the authors of the commits edit those?